File: Practica General/POO/Ejercicio 5/Estudiante.java
public class Estudiante {
    private String nombre;
    private double nota1;
    private double nota2;

    public Estudiante(String nombre, double nota1, double nota2) {
        this.nombre = nombre;
        this.nota1 = nota1;
        this.nota2 = nota2;
    }

    public double calcularPromedio() {
        return (nota1 + nota2) / 2;
    }

    public boolean verificarAprobado() {
        return calcularPromedio() >= 6;
    }

    public void mostrarInfo() {
        double promedio = calcularPromedio();
        String estado = verificarAprobado() ? "Aprobado" : "Reprobado";
        System.out.printf("%s - Promedio: %.2f - %s%n", nombre, promedio, estado);
    }

    public static void main(String[] args) {
        Estudiante e1 = new Estudiante("Lucía", 8, 7);
        Estudiante e2 = new Estudiante("Carlos", 5, 6);
        Estudiante e3 = new Estudiante("Ana", 4, 5);

        System.out.println("=== Información de Estudiantes ===");
        Estudiante[] estudiantes = {e1, e2, e3};

        for (Estudiante e : estudiantes) {
            e.mostrarInfo();
        }
    }
}

File: Practica General/POO/Ejercicio 5/estudiante.py
class Estudiante:
    def __init__(self, nombre, nota_1, nota_2):
        self.nombre = nombre
        self.nota_1 = nota_1
        self.nota_2 = nota_2

    def calcular_promedio(self):
        return (self.nota_1 + self.nota_2) / 2

    def verificar_aprobado(self):
        return self.calcular_promedio() >= 6

    def mostrar_info(self):
        promedio = self.calcular_promedio()
        estado = "Aprobado" if self.verificar_aprobado() else "Reprobado"
        print(f"{self.nombre} - Promedio: {promedio:.2f} - {estado}")


# Crear 3 estudiantes
e1 = Estudiante("Lucía", 8, 7)
e2 = Estudiante("Carlos", 5, 6)
e3 = Estudiante("Ana", 4, 5)

print("=== Información de Estudiantes ===")
for estudiante in [e1, e2, e3]:
    estudiante.mostrar_info()
